test/functional/createnewblock.py

```diff
@@ -0,0 +1,322 @@
+        mempool_entry = node.getmempoolentry(txid_to_bump)
+        ancestor_size = mempool_entry['ancestorsize']
+        ancestor_fee = mempool_entry['ancestorfees']
+        tx_fee = mempool_entry['modifiedfee']
+        tx_size = mempool_entry['size']
+        self.log.debug(
+                "Found txid to bump: %s, ancestor_size = %d, ancestor_fee = %f (ancestor feerate: %f) size = %d mod_fee = %d (feerate: %f)",
+            txid_to_bump, ancestor_size, ancestor_fee, ancestor_fee/ancestor_size, tx_size, tx_fee, tx_fee/tx_size)
+
+        # Determine feerate of the last ancestor_size portion of the block.
+        cumulative_weight = 0
+        max_feerate = 0
+        for block_tx in reversed(template["transactions"]):
+            # The fee is the consensus correct one, not the policy modified
+            # one, but we haven't prioritized anything yet, so this should be
+            # fine.
+            self.log.debug("End of block tx: txid %s, fee %d, weight %d",
+                    block_tx["txid"], block_tx["fee"], block_tx["weight"])
+            if 4 * block_tx["fee"] / block_tx["weight"] > max_feerate:
+                max_feerate = 4 * block_tx["fee"] / block_tx["weight"]
+            cumulative_weight += block_tx["weight"]
+            if cumulative_weight > 4 * ancestor_size:
+                break
+
+        # Bump the candidate transaction so that it has a higher feerate than
+        # the max feerate observed in the last portion of the block that we
+        # seek to replace
+        fee_bump = int(max_feerate * ancestor_size - ancestor_fee) + 1
+        self.log.debug("Setting fee_bump to %d", fee_bump)
+
+        node.prioritisetransaction(txid=txid_to_bump, fee_delta=fee_bump)
+
+        # Check that the next block template has txid_to_bump in it.
+        self.add_empty_block(node)  # bypass the gbt cache
+        template = node.getblocktemplate({"rules": ["segwit"]})
+        new_block_txids = [x["txid"] for x in template["transactions"]]
+
+        assert txid_to_bump in new_block_txids
+
+        # Get rid of our prioritisation
+        node.prioritisetransaction(txid=txid_to_bump, fee_delta=-fee_bump)
+
+    def test_recent_transactions(self):
+        # Populate the mempool with some transactions, and compare gbt results
+        # across the nodes.  Repeat.
+        self.add_empty_block(self.nodes[0])  # Bypass gbt cache.
+        self.sync_all()
+        recent_tx_block_count = [0, 0, 0]
+
+        max_iterations = 20
+        for i in range(max_iterations):
+            self.populate_mempool(random.randint(5, 15))  # 5-15 transactions
+
+            # Freeze current time, so that our recent tx window calculations
+            # will be consistent.
+            cur_time = int(time.time())
+            [node.setmocktime(cur_time) for node in self.nodes]
+
+            # Check that total block reward is within stale_rate of node3
+            max_income = self.nodes[3].getblocktemplate({
```

Comment:
I made the functional tests default to having -recenttxstalerate=0, to avoid breakage of existing tests due to this new behavior (see the change to `test_node.py` in https://github.com/bitcoin/bitcoin/pull/10200/commits/72e2ab0378c2baa13b6f1e80b3adcc302787c367#diff-86294e5ae5283eebdd9f98d79007a0e1).  I can add a comment as well...